File: revisi/app/src/main/java/com/ukom/resto2/adapters/DetailedDailyAdapter.java
package com.ukom.resto2.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.bottomsheet.BottomSheetDialog;
import com.ukom.resto2.R;
import com.ukom.resto2.models.DailyMealModel;
import com.ukom.resto2.models.DetailedDailyModel;

import java.util.List;

public class DetailedDailyAdapter extends RecyclerView.Adapter<DetailedDailyAdapter.ViewHolder> {

    private BottomSheetDialog bottomSheetDialog;
    List<DetailedDailyModel> list;

    public DetailedDailyAdapter(List<DetailedDailyModel> list) {
        this.list = list;
    }

    @NonNull
    @Override
    public DetailedDailyAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.detailed_daily_meal_item,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull DetailedDailyAdapter.ViewHolder holder, int position) {
        holder.imageView.setImageResource(list.get(position).getImage());
        holder.price.setText(list.get(position).getPrice());
        holder.name.setText(list.get(position).getName());
        holder.description.setText(list.get(position).getDescription());
        holder.timing.setText(list.get(position).getTiming());
        holder.rating.setText(list.get(position).getRating());

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        ImageView imageView;
        TextView name,price,description,rating,timing;
        Button button;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            imageView = itemView.findViewById(R.id.detailed_img);
            price = itemView.findViewById(R.id.detailed_name);
            name = itemView.findViewById(R.id.detailed_price);
            description = itemView.findViewById(R.id.detailed_des);
            rating = itemView.findViewById(R.id.detailed_rating);
            timing = itemView.findViewById(R.id.detailed_timing);
            button = itemView.findViewById(R.id.button);

            bottomSheetDialog = new BottomSheetDialog(button.getContext(),R.style.BottomSheetTheme);

            button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    Toast.makeText(button.getContext(), "Added to a Cart", Toast.LENGTH_SHORT).show();
                    bottomSheetDialog.dismiss();
                }
            });

        }
    }
}

File: revisi/app/src/main/java/com/ukom/resto2/adapters/DetailedDailyAdapter.kt
package com.ukom.resto2.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.bottomsheet.BottomSheetDialog
import com.ukom.resto2.R
import com.ukom.resto2.models.DetailedDailyModel

class DetailedDailyAdapter(var list: List<DetailedDailyModel>) :
    RecyclerView.Adapter<DetailedDailyAdapter.ViewHolder>() {
    private var bottomSheetDialog: BottomSheetDialog? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.detailed_daily_meal_item, parent, false)
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.imageView.setImageResource(list[position].image)
        holder.price.text = list[position].price
        holder.name.text = list[position].name
        holder.description.text = list[position].description
        holder.timing.text = list[position].timing
        holder.rating.text = list[position].rating
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var imageView: ImageView =
            itemView.findViewById(R.id.detailed_img)
        var name: TextView =
            itemView.findViewById(R.id.detailed_price)
        var price: TextView =
            itemView.findViewById(R.id.detailed_name)
        var description: TextView =
            itemView.findViewById(R.id.detailed_des)
        var rating: TextView =
            itemView.findViewById(R.id.detailed_rating)
        var timing: TextView =
            itemView.findViewById(R.id.detailed_timing)
        var button: Button =
            itemView.findViewById(R.id.button)

        init {
            bottomSheetDialog = BottomSheetDialog(button.context, R.style.BottomSheetTheme)

            button.setOnClickListener {
                Toast.makeText(button.context, "Added to a Cart", Toast.LENGTH_SHORT).show()
                bottomSheetDialog!!.dismiss()
            }
        }
    }
}
